test/general/test_options.py

```diff
@@ -10,3 +10,9 @@ def test_options_have_doc_string(self):
                 for option_key, option in world_type.options_dataclass.type_hints.items():
                     with self.subTest(game=gamename, option=option_key):
                         self.assertTrue(option.__doc__)
+
+    def test_options_are_not_set_by_world(self):
+        """Test that options attribute is not already set"""
+        for gamename, world_type in AutoWorldRegister.world_types.items():
+            with self.subTest(game=gamename):
+                self.assertFalse(hasattr(world_type, "options"))
```

Comment:
I would suggest putting an explicit message there
```suggestion
                self.assertFalse(hasattr(world_type, "options"),
                                 f"Unexpected assignment to {world_type.__name__}.options!")
```